### PR TITLE
Domain feature to authenticate ssh users with email in DUO

### PR DESCRIPTION
**Pull request by amanav82**
Adding domain option in pam_duo.conf file

This is to add 'domain' field to pam_duo.conf.
This is the working fix where DUO username is their email id instead of just username.
And when you are trying to configure ssh with DUO authentication, it tries to
register a new 'USER' with DUO even though 'USER@MYCOMPANY.com' is there in DUO.

With this feature, once  'USER' gets authenticated with pam_unix, we are apending
'domain' value to 'USER' and making it 'USER@MYCOMPANY.com' to get authenticated from DUO.

Example pam_duo.conf file:
[duo]
; Duo integration key
ikey = XXXXXXXXXXXXXXXXXX
; Duo secret key
skey = XXXXXXXXXXXXXXXXX
; Duo API host
host = api-XXXXXXXXXXXXXXXXXXXX
; Send command for Duo Push authentication
;pushinfo = yes
; authenticate email instead of username by appending domain
domain = mycompany.com

Now 'abc' user gets authenticated with unix as 'abc' and with DUO as 'abc@mycompany.com'

### CODE REVIEW

**Comment by cessner-nw** [comment]
we need to rebuild this package due to  a sec issue. Can we do that as well?

**Comment by amanav82** [comment]
@cessner-nw I rebuilt our custom package with latest source and installed already

File: pam_duo/pam_duo.c [hunk 1]
```diff
@@ -233,8 +233,17 @@ pam_sm_authenticate(pam_handle_t *pamh, int pam_flags,
     pam_err = PAM_SERVICE_ERR;
 
     for (i = 0; i < cfg.prompts; i++) {
-        code = duo_login(duo, user, host, flags,
-                    cfg.pushinfo ? cmd : NULL);
+        if (!cfg.domain || !cfg.domain[0]) {
+            code = duo_login(duo, user, host, flags,
+                cfg.pushinfo ? cmd : NULL);
+        } else {
+            char *duo_auth_user=(char*)malloc(strlen(user)+strlen(cfg.domain)+2);
```
Comment (BradleyHiggins) [inline pam_duo/pam_duo.c]: duo_auth_user is never free'd, and the malloc() return ought to be checked before use.

**Comment by BradleyHiggins** [comment]
Thanks for the PR! Unfortunately, we'd like to limit the amount of user name modification that can occur in Duo Unix. Please contact support@duo.com so we can explore other ways of solving this issue for you. (Or you may simply continue using your modified Duo Unix.)

Thanks for using Duo!

**Pull request closed without merging**